source/polygon.cpp

```diff
@@ -0,0 +1,46 @@
+ld perimeter(vector<Point<ll>>& poli){
+    ld res = 0.0;
+    int n = ssize(poli);
+    for(int i = 1; i <= n; ++i) res += poli[(i-1) % n].dist(poli[(i)%n]);
+    return res;
+}
+
+// returns 2*area; area < 0 if clock-wise, area > 0 if ccw
+ll double_area(vector<Point<ll>>& poli){
+    ll res = 0.0;
+    int n = ssize(poli);
+    for(int i = 0; i < n; ++i)
+        res += (poli[(i)%n].x * poli[(i+1)%n].y - poli[(i+1)%n].x*poli[(i)%n].y);
```

Comment:
`(i)%n` can be simply `i`, the modulo will never change `i`

Reply:
fixed